Variant.mpl

```diff
@@ -137,6 +138,29 @@ Variant: [{
 
   visit: [0 visitInternal];
 
+  equal: [
+    other:;
+
+    i: 0 static;
+    typeList other.typeList same [
+      typeTag other.typeTag = [
+        result: FALSE;
+        [
+          i other.typeList fieldCount < [
+            i other.typeTag = [
+              i get i other.get = @result set
+            ] when
+            i 1 + @i set
+            TRUE static
+          ] [
+            FALSE static
+          ] if
+        ] loop
+        result
+      ] &&
+    ] &&
+  ];
```

Comment:
Here, the loop is redundant.

1. Check if the type-lists are the same.
2. check if the type-tags are the same.
3. Check if the values are the same.
4. Done.

Reply:
I tried to access a value with a [typeTag], but since it's dynamic, I got an error "Dynamic index in a non-homogeneous combined". Am I missing something and there is a workaround for that?

Reply:
My bad, now I see.

I think the function should be used only with objects with the same type-lists. Also, we can use`times` instead of `loop` - but it is not consistent with the rest of the code.
```
equal: [
  other:;
  typeList other.typeList same ~ ["Type mismatch." raiseStaticError] when
  result: FALSE;
  typeTag other.typeTag = [
    typeList fieldCount [
      i typeTag = [i get i other.get = !result] when
    ] times
  ] when
  @result
];
```

Reply:
Or we can use a post-test loop, instead of the current one.

Reply:
Also, next snippet produce an error:
```
[
  a: (0 0.0) Variant;
  b: (0 0.0) Variant;
  0.0 1 @a.assign
  a b = _:;
] call
```
```
... error, [raiseStaticError], Wrong tag in Tagged Union!
...
... [.equal], called from here
... [=], called from here
```

Reply:
> ... error, [raiseStaticError], Wrong tag in Tagged Union!
I am also having this error when using Variants (usually, when assigning values). Manually setting a tag after an initialization fixes this error ([0 @b.setTag] in the given code), and I haven't done any further investigation. I believe there's an issue in other part(s) of Variant.

Reply:
I think this difference can be a cause of an error:
```
typeTag: 0 dynamic;
```
and
```
INIT: [
  0 @typeTag set
  rawInit
];
```
in [INIT], static value is set

Reply:
Well, consider using next code:
```
getUnchecked: [memory storageAddress swap @typeList @ addressToReference];

equal: [
  other:;
  typeList other.typeList same ~ ["Schema mismatch" raiseStaticError] when
  result: FALSE;
  typeTag other.typeTag = [
    i: 0 static;
    [
      i typeTag = [i getUnchecked i other.getUnchecked = !result] when
      i 1 + !i
      i typeList fieldCount <
    ] loop
  ] when
  result
];
```

Reply:
Thanks